math/src/polynom/multi_linear.rs

```diff
@@ -0,0 +1,244 @@
+use alloc::vec::Vec;
+use core::ops::Index;
+
+#[cfg(feature = "concurrent")]
+pub use rayon::prelude::*;
+use utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
+
+use crate::FieldElement;
+
+// MULTI-LINEAR POLYNOMIAL
+// ================================================================================================
+
+/// Represents a multi-linear polynomial.
+///
+/// The representation stores the evaluations of the polynomial over the boolean hyper-cube
+/// ${0 , 1}^ν$.
+#[derive(Clone, Debug, PartialEq)]
+pub struct MultiLinearPoly<E: FieldElement> {
+    evaluations: Vec<E>,
+}
+
+impl<E: FieldElement> MultiLinearPoly<E> {
+    /// Constructs a [`MultiLinearPoly`] from its evaluations over the boolean hyper-cube ${0 , 1}^ν$.
+    pub fn from_evaluations(evaluations: Vec<E>) -> Self {
+        if !evaluations.len().is_power_of_two() {
+            panic!("A multi-linear polynomial should have a power of 2 number of evaluations over the Boolean hyper-cube")
+        }
```

Comment:
nit: I would write this as `assert!(evaluations.len().is_power_of_two(), "A multi-linear...");`.

Reply:
Changed